rfcs/rdm-0080-bulk-importer.md

```diff
@@ -0,0 +1,104 @@
+Start Date: 2025-02-01
+- RFC PR: https://github.com/inveniosoftware/rfcs/pull/92
+- Authors: Aaron McCollough, Esteban G. Gabancho
+
+# Bulk importer
+
+## Summary
+
+Currently, InvenioRDM's only affordance for the bulk creation, import, and/or editing of records and files requires direct CLI-command-driven engagement with records and files APIs.
+
+GUI-based bulk importing and editing of records and files is a widely desired, highly useful feature, which will help to make the platform appealing to a much broader base of institutional users.
+
+The proposed feature is a beta version of a bulk importer for metadata (in CSV format) and associated files.
+
+## Motivation
+As an administrator user I want to ...
+- Upload many records (with their files) at once into my instance.
+- Select in which communities the records are publishes.
+- Determine if a group of records get new DOIs (minted during the process)
+- Update many records at once.
+- Delete many records at once.
+- See the status of past and current uploads.
+
+##  Design
+
+> [!info]
+> Most of this content is "inspired" by [CDS ILS importer](https://github.com/CERNDocumentServer/cds-ils/tree/master/cds_ils/importer) and [lycophron](https://github.com/plazi/lycophron), kudos to both teams!
+
+![](0080/uml_1.png)
+
+The two main object within the bulk engine will be the *Importer* and *ImporterRecord*
+
+The **Importer** is the object that will orchestrate the bulk job, i.e. the parent object. It will contain metadata to help identify each of the job, like `title` and `description`, and also general information regarding the bulk operation: overall `status` , the `mode` of the operation, etc.
+The **ImporterRecord**, on the other hand, will be object that stores information about each individual object that the bulk operation will work with (create/update/delete). This object will store the raw metadata as well as the transformed information, some information to show as columns on the user interface, like the presence of files, or the matched record in case of updates. It will also have some slots for messages/errors.
+
+This two object won't hold any functionality regarding processing the input content and/or creating objects inside Invenio RDM. Instead, they will be responsible for orchestrating the process: 
+- Create tasks for each action
+- Store output messages from the tasks
+- Hold log information about the actions
+- ...
+To enable easily search and files integration we believe these two objects should extend from records like many other modules do (collections, requests, etc). 
+
+The other two objects involved will be the *Resources* and the *Serializers*.
```

Comment:
We might want to change this name to avoid confusion with `resources` and `services`. Any suggestions? ☺️

Reply:
what do you think about `record type` instead of resource? I think we used this name elsewhere too